images/bootstrap/runner.sh

```diff
@@ -90,6 +90,7 @@ mkdir -p "${GOPATH}/bin"
 # Authenticate gcloud, allow failures
 if [[ -n "${GOOGLE_APPLICATION_CREDENTIALS:-}" ]]; then
   gcloud auth activate-service-account --key-file="${GOOGLE_APPLICATION_CREDENTIALS}" || true
+  gcloud auth configure-docker
```

Comment:
uh, what are we pushing here?
normally images being pushed have specific credentials etc. why should this be in bootstrap?

Reply:
Updated the initial comment with rationale.
At the moment if e.g. kube-build is stored in a private registry, and Dockerfile is patched, then pull would fail even though the service account configured for GCS access has all the necessary permissions.

Placing it in bootstrap runner makes jobs with GOOGLE_APPLICATION_CREDENTIALS also have docker use these without any additional configuration. Do you believe it is better to be left to the individual image build processes instead?

Reply:
this image is not really intended for general usage / private registries? it's just something we build for testing the kubernetes project.

we also may not be doing anything with docker in many of the jobs.

Reply:
(is there any side effects?)

Reply:
Similar behavior guarded by `GOOGLE_APPLICATION_CREDENTIALS` exists in 
* https://github.com/kubernetes/test-infra/blob/master/kubetest/main.go#L845
* https://github.com/kubernetes/test-infra/blob/master/prow/bump.sh#L71

I still believe this configuration should be in runner, and enabled via `GOOGLE_APPLICATION_CREDENTIALS` in the runner, to completely configure all the supported utilities relevant to the build/test within the container to use the service account as specified in the prow job configuration.

Re: side effects - the `gcoud auth configure-docker` would place the `~/.docker/config.json` and cause authenticated access to gcr.io. Given that the registries currently referenced are public, there will be no change in behavior.